src/main/java/gregtech/integration/jei/recipe/GTRecipeWrapper.java

```diff
@@ -108,7 +108,7 @@ public void addTooltip(int slotIndex, boolean input, Object ingredient, List<Str
 
     @Override
     public void drawInfo(Minecraft minecraft, int recipeWidth, int recipeHeight, int mouseX, int mouseY) {
-        minecraft.fontRenderer.drawString(I18n.format("gregtech.recipe.total", Math.abs(recipe.getEUt()) * recipe.getDuration()), 0, 70, 0x111111);
+        minecraft.fontRenderer.drawString(I18n.format("gregtech.recipe.total", Math.abs((long) recipe.getEUt()) * recipe.getDuration()), 0, 70, 0x111111);
```

Comment:
What is this casting accomplishing?  Precision?

Reply:
we have 1 integer multiply by 1 integer the result is an Integer, so 2^31 max and when I make UV recipe I reach that point. 
By casing the function into long I can multiply 1 Long to 1 Integer the result can be 2^63.

Reply:
See [this](https://github.com/GregTechCE/GregTech/issues/1068) issue.